readme.md

```diff
@@ -15,121 +15,180 @@ Operations
+this version uses the identifier provided by the client rather than generating a new id.
 
-Accepts a POST request with ```path```, ```identifier``` and ```system``` parameters. This operation will create a new path entry for the path
-requested iff the the path is not currently in use. The user supplied numeric id for to identify the object that the path links to.
+To register an existing path issue a PUT request to ```/paths/<id>``` with the path record data as json in the body:
+
+```
+    {
+        "path":"<path>",
+        "identifier":<id>,
+        "type":"canonical", // canonical paths can be registered currently
+        "system":"<system>"
+    }
+```
 
-If successful this operation will return a JSON response with the paths registered, These are indexed by path type.
+If successful this operation will return an argo response with the paths registered, These are indexed by path type.
 
 example:
 
 ```
-    curl --data "path=foo/bar/baz1&identifier=345&system=test" https://pathmanager.local.dev-gutools.co.uk/registerPath
+    curl -i -XPUT -H"Content-Type: application/json" -d '{"path":"foo/bar/baz1","identifier":345,"type":"canonical","system":"test"}' https://pathmanager.local.dev-gutools.co.uk/paths/345
 ```
 
 returns
 
 ```
-    {
-    "canonical": {"path":"foo/bar/baz1","identifier":345,"type":"canonical","system":"test"},
-    "short":{"path":"simulatedShort/foo/bar/baz1","identifier":345,"type":"short","system":"test"}
+    {data: 
+        {"canonical":
+            [{
+            "path":"foo/bar/baz1",
+            "identifier":345,
+            "type":"canonical",
+            "system":"test"
+            }],
+        "short":
+            [{
+            "path":"simulatedShort/foo/bar/baz1",
+            "identifier":345,
+            "type":"short",
+            "system":"test"
+            }]
+        }
     }
 ```
 
 
-/updateCanonicalPath
---------------------
+Update a canonical path
+----------------------
 
-This endpoint is used to update the path that an item is on. It accepts a POST request with ```newPath```, ```existingPath```
-and ```identifier``` parameters. If the new path is available and the existing path corresponds to the provided identifier
-then the old path entry is removed and the new record with the new path is inserted.
+To update a canonical path for an item issue a POST request to ```/paths/<id>``` with ```newPath``` parameter.
+If the new path is available then the old path entry is removed and the new record with the new path is inserted.
 
-If successful this operation will return a JSON response with the path record
+If successful this operation will return a argo json response with the updated canonical path record
 
 example:
 
 ```
-    curl --data "newPath=foo/bar/hux&existingPath=foo/bar/baz1&identifier=345" https://pathmanager.local.dev-gutools.co.uk/updateCanonicalPath
+    curl --data "newPath=foo/bar/hux" https://pathmanager.local.dev-gutools.co.uk/paths/345
```

Comment:
Can we just send `path`? We discussed JSON vs form-encoded, I don't mind for now.